### PR TITLE
Provide an option to disable HTML emails

### LINKED ISSUE #9031: Implement "only plain-text e-mail" option
We are using Nextcloud with a GPG encrypting SMTP proxy ([kuvert](https://github.com/az143/kuvert)). HTML e-mail is notoriously problematic with GPG encryption, and things break especially when there are attachments added to the mix (like calendar invites).

It would be a very welcome feature if it were possible to check a check-box somewhere in e-mail settings to instruct Nextcloud to *only* sent plain-text e-mails (no HTML what-so-ever).

### PR DESCRIPTION
**Pull request by MorrisJobke**
Needs to be back ported to stable12 to fix an issue with IBM notes (see #9102).

Unit tests are included.

Fixes #9031

### CODE REVIEW

**Comment by MariusBluem** [comment]
😘😅

File: config/config.sample.php [hunk 1]
```diff
@@ -390,6 +390,18 @@
...
+ * options to modify the mail texts with the theming app is not enough.
+ * The class must extend  ``\OC\Mail\EMailTemplate``
+ */
+'mail_template_class' => '\OC\Mail\EMailTemplate',
+
+/**
+ * Email will be send by default with an HTML and a plain text body. This option
+ * allows to only send plain text emails.
+ */
+'mail_send_plaintext_only' => false,
```
Comment (MariusBluem) [inline config/config.sample.php]: Does it make sense to set a `mail_template_class` if `mail_send_plaintext_only` is `true`? Does this even work? If no - could we merge those two settings together?

**Review by rullzer — APPROVED** [review]
Makes sense

File: config/config.sample.php (hunk 1 above)
Comment (nickvergessen) [reply config/config.sample.php]: yes, you can still replace text with the template class.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**